vault/audit_broker.go

```diff
@@ -63,43 +69,30 @@ func (a *AuditBroker) Register(name string, b audit.Backend, local bool) error {
 	a.Lock()
 	defer a.Unlock()
 
-	a.backends[name] = backendEntry{
-		backend: b,
-		local:   local,
+	name = strings.TrimSpace(name)
+	if name == "" {
+		return fmt.Errorf("%s: name is required: %w", op, event.ErrInvalidParameter)
 	}
 
+	// The reason for this check is due to 1.15.x supporting the env var:
+	// 'VAULT_AUDIT_DISABLE_EVENTLOGGER'
+	// When NewAuditBroker is called, it is supplied a bool to determine whether
+	// we initialize the broker, which are left nil otherwise.
+	// In 1.16.x this check should go away and the env var removed.
 	if a.broker != nil {
 		if name != b.Name() {
 			return fmt.Errorf("%s: audit registration failed due to device name mismatch: %q, %q", op, name, b.Name())
 		}
 
-		for id, node := range b.Nodes() {
-			err := a.broker.RegisterNode(id, node, eventlogger.WithNodeRegistrationPolicy(eventlogger.DenyOverwrite))
-			if err != nil {
-				return fmt.Errorf("%s: unable to register nodes for %q: %w", op, name, err)
-			}
-		}
-
-		pipeline := eventlogger.Pipeline{
-			PipelineID: eventlogger.PipelineID(b.Name()),
-			EventType:  b.EventType(),
-			NodeIDs:    b.NodeIDs(),
-		}
-
-		err := a.broker.RegisterPipeline(pipeline, eventlogger.WithPipelineRegistrationPolicy(eventlogger.DenyOverwrite))
+		err := a.register(name, b)
 		if err != nil {
-			return fmt.Errorf("%s: unable to register pipeline for %q: %w", op, name, err)
+			return fmt.Errorf("%s: unable to register device for %q: %w", op, name, err)
 		}
+	}
 
-		// Establish if we ONLY have pipelines that include filter nodes.
-		// Otherwise, we can rely on the eventlogger broker guarantee.
-		threshold := a.requiredSuccessThresholdSinks()
-
-		// Update the success threshold now that the pipeline is registered.
-		err = a.broker.SetSuccessThresholdSinks(eventlogger.EventType(event.AuditType.String()), threshold)
-		if err != nil {
-			return fmt.Errorf("%s: unable to configure sink success threshold (%d) for %q: %w", op, threshold, name, err)
-		}
+	a.backends[name] = backendEntry{
+		backend: b,
+		local:   local,
```

Comment:
Adding to the backend map was moved to the bottom of the function so that we never think we've got a backend that failed higher up in this func.